src/components/SideNavBar/SideNavBarTop.jsx

```diff
@@ -8,27 +8,71 @@ type Props = {|
   label: string,
   icon: string,
   onClick: () => void,
+  editing: boolean,
+  placeholder: string,
+  value: string,
+  onChange: ({ value: string }) => void,
+  onBlur: () => void,
+  fullWidth: boolean,
+  className: string,
+  forwardedRef?: { current: null | HTMLInputElement },
```

Comment:
I believe there is a `React.Ref` type.

Reply:
Until the upcoming version of flow, that type doesn't reflect react 16's new createRef API (e.g., the ref does not have a `current` attribute). This is the recommended workaround.